CMakeLists.txt

```diff
@@ -49,6 +49,185 @@ if(GNU_GCC)
   add_compile_options(-pipe)
 endif()
 
+option(WARNINGS_PEDANTIC "Let the compiler show even more warnings" OFF)
+if(MSVC)
+  if(WARNINGS_PEDANTIC)
+    add_compile_options(/W4)
+  else()
+    add_compile_options(/W3)
+  endif()
+else()
+  add_compile_options(-Wall -Wextra)
+  if(WARNINGS_PEDANTIC)
+    add_compile_options(-pedantic)
+  endif()
+endif()
+
+option(WARNINGS_FATAL "Fail if compiler generates a warning" OFF)
+if(WARNINGS_FATAL)
+  if(MSVC)
+    add_compile_options(/WX)
+  else()
+    add_compile_options(-Werror)
```

Comment:
Won't this make the build fail if one of your dependencies in `lib/` folder contain a warning? I think we should keep this restricted to `mixxx-lib`.

Reply:
Yes, it fails by design, isn't it?

Reply:
It should fail because there is a warning in Mixxx, not in some dependency like gtest.